scripts/ui/mode_selection.py

```diff
@@ -0,0 +1,170 @@
+import bpy
+
+class ARF_OT_Armature_Mode(bpy.types.Operator):
+    bl_label = "Armature Mode"
+    bl_idname = "arf.armature_mode"
+
+    def execute(self, context):
+        for obj in context.scene.objects:
+            if (obj.type == "ARMATURE"):
+                obj.hide_set(False)
+            else:
+                obj.hide_set(True)
+        return {'FINISHED'}
+
+
+class ARF_OT_Skeleton_Mode(bpy.types.Operator):
+    bl_label = "Skeleton Mode"
+    bl_idname = "arf.skeleton_mode"
+
+    def execute(self, context):
+        for obj in context.scene.objects:
```

Comment:
Please make sure to check if `obj.ARF` exists before using it, with an assertion. `obj.ARF` should in principle always be present, hence asserting, but this way `mypy` won't complain and it allows us to verify that the code works as intended.
```suggestion
        for obj in context.scene.objects:
            assert hasattr(obj, 'ARF')
```